internal/driver/ipam.go

```diff
@@ -141,137 +142,154 @@ func (i IpamDriver) ReleasePool(request *ipam.ReleasePoolRequest) error {
 	return nil
 }
 
+// RequestAddress .
 func (i IpamDriver) RequestAddress(request *ipam.RequestAddressRequest) (*ipam.RequestAddressResponse, error) {
 	logutils.JSONMessage("RequestAddress", request)
 
-	hostname, err := osutils.GetHostname()
-	if err != nil {
+	// Calico IPAM does not allow you to choose a gateway.
+	if request.Options["RequestAddressType"] == "com.docker.network.gateway" {
```

Comment:
defines as a const may be a better idea.

Reply:
legency code, but I think using literal will be much more readable here as the value will not be reused